### PR TITLE
issue #210 : use store values, instead of multiple api calls

### PR DESCRIPTION
**Pull request by richard-francois**
-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/jhipster-vuejs/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/jhipster-vuejs/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by richard-francois** [comment]
This is my first contribution / PR ! ... :-)

File: generators/client/templates/vue/src/main/webapp/app/account/principal.ts.ejs [hunk 1]
```diff
@@ -34,6 +34,16 @@ export default class Principal {
     if (!this.store.getters.account && !this.store.getters.logon && token) {
       this.retrieveAccount();
     }
+    this.retrieveProfiles();
+  }
+
+  public retrieveProfiles(): any {
+    axios.get('management/info').then((res) => {
+    if (res.data && res.data.activeProfiles && res.data.activeProfiles.indexOf(res.data['display-ribbon-on-profiles']) > -1) {
```
Comment (pascalgrimaud) [inline generators/client/templates/vue/src/main/webapp/app/account/principal.ts.ejs]: indent: need 2 additional spaces

**Comment by pascalgrimaud** [comment]
Congrats for your 1st contribution to the project!

For your information, you could add `fix https://github.com/jhipster/jhipster-vuejs/issues/210` in comment of your pull request, so the ticket will be closed automatically as soon as it's merged.

File: generators/client/templates/vue/src/main/webapp/app/account/principal.ts.ejs (hunk 1 above)
Comment (sahbi-ktifa) [reply generators/client/templates/vue/src/main/webapp/app/account/principal.ts.ejs]: Yes please, fix indentation and then we could merge this !

File: generators/client/templates/vue/src/test/javascript/spec/app/core/ribbon/ribbon.component.spec.ts.ejs [hunk 2]
```diff
@@ -46,25 +42,9 @@ describe('Ribbon', () => {
...
-    await wrap({ data: {} });
-    expect(ribbon.ribbonEnv).toBeNull();
-  });
-
-  it('should not have ribbonEnv when empty activeProfiles', async () => {
-    await wrap({ data: { activeProfiles: [] } });
-    expect(ribbon.ribbonEnv).toBeNull();
-  });
-
   it('should not have ribbonEnv when empty activeProfiles', async () => {
```
Comment (sahbi-ktifa) [inline generators/client/templates/vue/src/test/javascript/spec/app/core/ribbon/ribbon.component.spec.ts.ejs]: Please, can you also fix the test title as it was wrong from the beginning (copy / paste from test above I guess) ?

File: generators/client/templates/vue/src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts.ejs [hunk 3]
```diff
@@ -19,14 +19,12 @@ export default class JhiNavbar extends Vue {
...
-      }
-      if (res.data && res.data.activeProfiles && res.data.activeProfiles.indexOf('prod') > -1) {
-        this.inProduction = true;
-      }
-    });
+    if (this.activeProfiles.indexOf('swagger') > -1) {
+      this.swaggerEnabled = true;
+    }
+    if (this.activeProfiles.indexOf('prod') > -1) {
+      this.inProduction = true;
```
Comment (sahbi-ktifa) [inline generators/client/templates/vue/src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts.ejs]: Are you sure this is set properly ? Should not it be set using computed methods ?

File: generators/client/templates/vue/src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts.ejs [hunk 4]
```diff
@@ -74,4 +72,8 @@ export default class JhiNavbar extends Vue {
   public hasAnyAuthority(authorities: any): boolean {
     return this.principal().hasAnyAuthority(authorities);
   }
+
+  public get activeProfiles() : string {
+    return this.$store.getters.activeProfiles;
```
Comment (sahbi-ktifa) [inline generators/client/templates/vue/src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts.ejs]: ```activeProfiles``` do not need to be exposed as computed method as we do not use it directly, it seems to me that we should have here two computed methods (```swaggerEnabled``` / ```inProduction```), both using ```activeProfiles``` property from the store. 
I may be wrong but could you check database console is available in dev mode ? If not, that could be your solution.

File: generators/client/templates/vue/src/test/javascript/spec/app/core/ribbon/ribbon.component.spec.ts.ejs [hunk 5]
```diff
@@ -46,25 +42,9 @@ describe('Ribbon', () => {
     expect(ribbon.ribbonEnv).toBeNull();
   });
 
-  it('should not have ribbonEnv when no activeProfiles', async () => {
-    await wrap({ data: {} });
-    expect(ribbon.ribbonEnv).toBeNull();
-  });
-
-  it('should not have ribbonEnv when empty activeProfiles', async () => {
```
Comment (hdurix) [inline generators/client/templates/vue/src/test/javascript/spec/app/core/ribbon/ribbon.component.spec.ts.ejs]: Please move the deleted tests to principal test

Comment (richard-francois) [reply generators/client/templates/vue/src/test/javascript/spec/app/core/ribbon/ribbon.component.spec.ts.ejs]: writing the test, as it does not exists !

File: generators/client/templates/vue/src/test/javascript/spec/app/core/ribbon/ribbon.component.spec.ts.ejs (hunk 2 above)
Comment (richard-francois) [reply generators/client/templates/vue/src/test/javascript/spec/app/core/ribbon/ribbon.component.spec.ts.ejs]: done

File: generators/client/templates/vue/src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts.ejs (hunk 4 above)
Comment (richard-francois) [reply generators/client/templates/vue/src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts.ejs]: done

File: generators/client/templates/vue/src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts.ejs (hunk 3 above)
Comment (richard-francois) [reply generators/client/templates/vue/src/main/webapp/app/core/jhi-navbar/jhi-navbar.component.ts.ejs]: done

File: generators/client/templates/vue/src/main/webapp/app/account/principal.ts.ejs (hunk 1 above)
Comment (richard-francois) [reply generators/client/templates/vue/src/main/webapp/app/account/principal.ts.ejs]: done

File: generators/client/templates/vue/src/test/javascript/spec/app/core/ribbon/ribbon.component.spec.ts.ejs (hunk 5 above)
Comment (sahbi-ktifa) [reply generators/client/templates/vue/src/test/javascript/spec/app/core/ribbon/ribbon.component.spec.ts.ejs]: Please don't, principal will be refactor for oauth2 support, I'll do the test for this one. And yes, principal is indeed currently missing.

Comment (richard-francois) [reply generators/client/templates/vue/src/test/javascript/spec/app/core/ribbon/ribbon.component.spec.ts.ejs]: ok

**Pull request closed without merging**